crates/sui-framework/sources/governance/validator.move

```diff
@@ -593,9 +584,41 @@ module sui::validator {
         &self.staking_pool
     }
 
+    /// Create a new validator from the given `ValidatorMetadata`, called by both `new` and `new_for_testing`.
+    fun new_from_metadata(
+        metadata: ValidatorMetadata,
+        stake: Balance<SUI>,
+        coin_locked_until_epoch: Option<EpochTimeLock>,
+        gas_price: u64,
+        commission_rate: u64,
+        starting_epoch: u64,
+        ctx: &mut TxContext
+    ): Validator {
+        let sui_address = metadata.sui_address;
+        let stake_amount = balance::value(&stake);
+
+        let staking_pool = staking_pool::new(starting_epoch, ctx);
+        // Add the validator's starting stake to the staking pool.
+        staking_pool::request_add_delegation(&mut staking_pool, stake, coin_locked_until_epoch, sui_address, sui_address, starting_epoch, ctx);
```

Comment:
Is this function now shared between both delegation and self staking? If so, might want to rename it

Reply:
If we indeed decide to replace "delegation" with "stake" everywhere (adding the task to our tracker...), it should be in a later and much bigger PR.